docs/cluster.md

```diff
@@ -0,0 +1,195 @@
+
+:::tip Distribution Environment Variables
+
+List of all cluster-related environment variables is available [here](./getting_started/installation#general-env-variables).
+
+:::
+
+:::caution Extra network Configuration
+
+Jellyfish leverages Erlang Distribution to form a cluster.
+This means that we don't need to use any database where we would store
+information about network topology.
+Instead, some extra network configuration might be needed.
+
+* Jellyfish uses a service called EPMD (Erlang Port Mapper Deamon)
+that runs on port `4369` (TCP).
+If you run Jellyfish using Docker, you have to explicitly export this port.
+In production deployment, you also have to allow for traffic on this port in your firewall.
+EPMD is started automatically when running Jellyfish, so you don't need to take
+any extra steps to make it work.
+
+* Jellyfish uses a random port for connecting to other Jellyfishes.
+If you run Jellyfish using Docker, this port defaults to `9000` (TCP) but you can configure
+the range it is selected from with `JF_DIST_MIN_PORT` and `JF_DIST_MAX_PORT`.
+Note that `JF_DIST_MIN_PORT` and `JF_DIST_MAX_PORT` are not available when running from source 
+(using `mix phx.server`) as in development you don't have to worry about the port used for
+forming a cluster.
+As in the case of EPMD, in production deployment, you have to modify your firewall rules appropriately.
+
+See [Deeper dive into Erlang Distribution](#deeper-dive-into-erlang-distribution) for more information.
+:::
+
+
+
+## Examples
+
+
+### Running from source
+
+Run the first Jellyfish:
+
+```sh
+JF_DIST_ENABLED=true JF_DIST_NODE_NAME=j1@127.0.0.1 mix phx.server
+```
+
+Run the second Jellyfish
+
+```sh
+JF_DIST_ENABLED=true JF_DIST_NODE_NAME=j2@127.0.0.1 JF_DIST_NODES="j1@127.0.0.1" JF_PORT=4002 JF_METRICS_PORT=9468 mix phx.server
+```
+
+:::info
+
+Note that when running the second Jellyfish, we had to change its HTTP and metrics ports.
+
+:::
+
+
+### Running with Docker (locally)
+
+This simple docker compose file sets a cluster of two Jellyfishes.
+
+```yml
+version: "3"
+
+x-jellyfish-template: &jellyfish-template
+  build: .
+  environment: &jellyfish-environment
+    JF_SERVER_API_TOKEN: "development"
+    JF_DIST_ENABLED: "true"
+    JF_DIST_MODE: "sname"
+    JF_DIST_NODES: "j1@jellyfish1 j2@jellyfish2"
+  networks:
+    - net1
+  restart: on-failure
+
+services:
+  jellyfish1:
+    <<: *jellyfish-template
+    environment:
+      <<: *jellyfish-environment
+      JF_HOST: "localhost:5001"
+      JF_PORT: 5001
+      JF_METRICS_PORT: 6001
+      JF_DIST_NODE_NAME: j1@jellyfish1
+    ports:
+      - 5001:5001
+      - 6001:6001
+
+  jellyfish2:
+    <<: *jellyfish-template
+    environment:
+      <<: *jellyfish-environment
+      JF_HOST: "localhost:5002"
+      JF_PORT: 5002
+      JF_METRICS_PORT: 6002
+      JF_DIST_NODE_NAME: j2@jellyfish2
+    ports:
+      - 5002:5002
+      - 6002:6002
+
+networks:
+  net1:
+    driver: bridge
```

Comment:
I think that this is not needed because docker-compose by default creates a network that connects all services in this compose.

Reply:
Will check that

Reply:
Yup, looks like it's not needed :+1: